tee-worker/ts-tests/common/utils/context.ts

```diff
@@ -0,0 +1,154 @@
+import { ethers } from 'ethers';
+import WebSocketAsPromised from 'websocket-as-promised';
+import WebSocket from 'ws';
+import Options from 'websocket-as-promised/types/options';
+import { KeyObject } from 'crypto';
+import { getMetadata } from '../call';
+import { getEthereumSigner, getSubstrateSigner } from '../helpers';
+import { IntegrationTestContext, teeTypes, EnclaveResult, Web3Wallets } from '../type-definitions';
+import { decryptWithAES } from './crypto';
+
+const crypto = require('crypto');
+
+// maximum milliseconds that we wait in listening events before we timeout
+const listenTimeoutInMilliSeconds = 3 * 60 * 1000;
+
+export async function getListenTimeoutInBlocks(api: ApiPromise) {
+    const slotDuration = await api.call.auraApi.slotDuration();
+    return listenTimeoutInMilliSeconds / parseInt(slotDuration.toString());
+}
+
+export async function initWorkerConnection(endpoint: string): Promise<WebSocketAsPromised> {
+    const wsp = new WebSocketAsPromised(endpoint, <Options>(<unknown>{
+        createWebSocket: (url: any) => new WebSocket(url),
+        extractMessageData: (event: any) => event,
+        packMessage: (data: any) => JSON.stringify(data),
+        unpackMessage: (data: string | ArrayBuffer | Blob) => JSON.parse(data.toString()),
+        attachRequestId: (data: any, requestId: string | number) => Object.assign({ id: requestId }, data),
+        extractRequestId: (data: any) => data && data.id, // read requestId from message `id` field
+    }));
+    await wsp.open();
+    return wsp;
+}
+
+export async function initIntegrationTestContext(
+    workerEndpoint: string,
+    substrateEndpoint: string,
+    walletsNumber: number
+): Promise<IntegrationTestContext> {
+    const provider = new WsProvider(substrateEndpoint);
+    const ethersWallet = {
+        alice: new ethers.Wallet(getEthereumSigner().alice),
+        bob: new ethers.Wallet(getEthereumSigner().bob),
+        charlie: new ethers.Wallet(getEthereumSigner().charlie),
+        dave: new ethers.Wallet(getEthereumSigner().dave),
+        eve: new ethers.Wallet(getEthereumSigner().eve),
+    };
+    const substrateWallet = {
+        alice: getSubstrateSigner().alice,
+        bob: getSubstrateSigner().bob,
+        charlie: getSubstrateSigner().charlie,
+        eve: getSubstrateSigner().eve,
+    };
+
+    const { types } = teeTypes;
+    const api = await ApiPromise.create({
+        provider,
+        types,
+    });
+
+    await cryptoWaitReady();
+
+    const wsp = await initWorkerConnection(workerEndpoint);
+
+    const metaData = await getMetadata(wsp, api);
+
+    const web3Signers = await generateWeb3Wallets(walletsNumber);
+    const { mrEnclave, teeShieldingKey } = await getEnclave(api);
+    return <IntegrationTestContext>{
+        tee: wsp,
+        api,
+        teeShieldingKey,
+        mrEnclave,
+        ethersWallet,
+        substrateWallet,
+        metaData,
+        web3Signers,
+    };
+}
+
+export async function handleVcEvents(
```

Comment:
We have `identity.ts` now. Would it be better to move it into `vc.ts`? Additionally, wouldn't it be clearer to name them `identity_helper` and `vc_helper`?